src/main/java/net/openhft/chronicle/threads/MediumEventLoop.java

```diff
@@ -520,12 +521,18 @@ protected synchronized boolean updateHighHandler(@NotNull EventHandler handler)
     @Override
     public void dumpRunningState(@NotNull final String message, @NotNull final BooleanSupplier finalCheck) {
         final Thread threadSnapshot = this.thread;
-        if (threadSnapshot == null) return;
+        if (threadSnapshot == null || !Jvm.isPerfEnabled(getClass()))
+            return;
         final StringBuilder out = new StringBuilder(message);
+        final int messageIndex = out.length();
+        final long startTimeNanos = System.nanoTime();
         Jvm.trimStackTrace(out, threadSnapshot.getStackTrace());
 
-        if (finalCheck.getAsBoolean() && Jvm.isDebugEnabled(getClass()))
-            Jvm.debug().on(getClass(), out.toString());
+        if (finalCheck.getAsBoolean()) {
+            final double timeToTakeStackTraceMillis = Maths.round1((System.nanoTime() - startTimeNanos) / 1e6);
+            out.insert(messageIndex, " (below stack trace may be inaccurate - capturing stack trace took " + timeToTakeStackTraceMillis + "ms)");
```

Comment:
Personally I think this is an improvement. It is better to be notified that a blockage was detected than the detection be suppressed, and less surprising for the user that all blockages are logged.